Allow more default sink parameters

Lokdei opened the pull request:
Hi

This PR allows to add sink parameters 
- restrictedToMinimumLevel
- outputTemplate
- formatProvider 
- levelSwitch

And 1 test to check if the restrictedToMinimumLevel works as expected.

Would you please give feedback upon this and consider merging it ?

The tests all pass.

sandermvanvliet commented:
Hi @lokdei thanks for your PR!
I’ll give this a look over the weekend

sandermvanvliet commented on src/Serilog.Sinks.InMemory/InMemorySink.cs:
```diff
@@ -3,12 +3,14 @@
     public class InMemorySink : ILogEventSink, IDisposable
     {
         private static readonly AsyncLocal<InMemorySink> LocalInstance = new AsyncLocal<InMemorySink>();
+        private readonly ITextFormatter _formatProvider;
```
Given that the `_formatProvider` is never used I would recommend removing it, that would also remove the necessity for `GetInstance()` as you can use `Instance` like before.

sandermvanvliet commented on src/Serilog.Sinks.InMemory/InMemorySinkExtensions.cs:
```diff
@@ -1,13 +1,65 @@
+            if (sinkConfiguration == null) throw new ArgumentNullException(nameof(sinkConfiguration));
+            if (formatter == null) throw new ArgumentNullException(nameof(formatter));
+
+            return sinkConfiguration.Sink(InMemorySink.GetInstance(formatter), restrictedToMinimumLevel, levelSwitch);
```
As mentioned in the other comment, I don't think you need to supply the `formatter` here as it isn't used by the sink.

Lokdei commented:
Hey

I added a section in the Readme: "Creating a logger" with setup parameter examples.

I also removed the unused `formatter` from the constructor and GetInstance() call.

sandermvanvliet commented:
@Lokdei  version [0.5.0](https://www.nuget.org/packages/Serilog.Sinks.InMemory/0.5.0) has been pushed to NuGet and should be visible shortly
